src/tools/build_helper/src/util.rs

```diff
@@ -0,0 +1,33 @@
+use std::process::Command;
+
+/// If code is not 0 (successful exit status), exit status is 101 (rust's default error code.)
+/// If the test is running and code is an error code, it will cause a panic.
+pub fn detail_exit(code: i32, is_test: bool) -> ! {
```

Comment:
1- `cfg!(test)` will not work inside of the body because it's under `build_helper`. https://github.com/rust-lang/rust/actions/runs/5115125346/jobs/9196077533 This is why CI was failed

2- If this was called from testing frameworks like `compiletest` or `rustdoc-gui-test`, cfg test will not help there too.

Reply:
I created `detail_exit_macro` helper which invokes `detail_exit` with `cfg!(test)` that solves `cfg!(test) will not work inside of the body because it's under build_helper` this thing,

and remain the `detail_exit` function same which will be useful for ` If this was called from testing frameworks like compiletest or rustdoc-gui-test, cfg test will not help there too.` this use cases.